### ISSUE TITLE
Comment support?

### ISSUE THREAD

**Issue by MrFishCakes** [Carleslc/Simple-YAML#4]
Hi, I don't have an issue at all I was just wondering if this API supports comments that are made in the YML file? I know that by default SnakeYAML doesn't handle these and I was wondering if this had been implemented into this API?

**Comment by Carleslc** [comment]
Yes, with some considerations.

You can load a file with comments inside, make changes and then save your YamlFile using method `saveWithComments` instead usual `save`. This is more time-consuming than `save`, so if you don't need comments `save` is recommended.

Note that comments to the right of key/values are NOT copied, only those lines starting with # character, ignoring tabs and spaces:

```
# this comment will be copied
number: 2 # this comment will NOT be copied
```

**Comment by Carleslc** [comment]
A new release is available ([1.5](https://github.com/Carleslc/Simple-YAML/releases/tag/1.5)) with improved support for comments.

Side comments are now supported.

```yml
# this comment will be copied
number: 2 # this comment now will be copied too
```

Note, though, when going to use comments you must use `saveWithComments` instead of `save`. In addition, when adding comments via code `loadWithComments` should be used too instead of `load` to preserve comments that may already exist.
